src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,719 @@
+        super(packetPredicate);
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * The (default) implementation of {@code SinglePacketTransformer} does
+     * nothing.
+     */
+    @Override
+    public void close() {
+        super.close();
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Reverse-transforms an array of packets by calling
+     * {@link #reverseTransform(RawPacket)} on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] reverseTransform(RawPacket[] pkts) {
+        return super.reverseTransform(pkts);
+    }
+
+    /**
+     * Transforms a specific packet.
+     *
+     * @param pkt the packet to be transformed.
+     * @return the transformed packet.
+     */
+    @Override
+    public RawPacket transform(RawPacket pkt) {
+        return null;
+    }
+
+    /**
+     * {@inheritDoc}
+     * <p>
+     * Transforms an array of packets by calling {@link #transform(RawPacket)}
+     * on each one.
+     *
+     * @param pkts
+     */
+    @Override
+    public RawPacket[] transform(RawPacket[] pkts) {
+        return super.transform(pkts);
+    }
+
+    /**
+     * Returns the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}. The method is called on the same
+     * worker thread as Process will be called on.
+     *
+     * @return the number of milliseconds until this instance wants a worker
+     * thread to call {@link #run()}
+     */
+    @Override
+    public long getTimeUntilNextRun() {
+        return 0;
+    }
+
+    private enum ProbeResult
+    {
+        kBitrateUpdated(0),
+        kNoUpdate(1);
+        int value;
+        ProbeResult(int x)
+        {
+            this.value = x;
+        };
+    }
+
+    private final Object critSect = new Object();
+    private Collection<Integer> ssrcs = new ArrayList<Integer>();
+    private final static int kTimestampGroupLengthMs = 5;
+    private final static int kAbsSendTimeFraction = 18;
+    private final static int kAbsSendTimeInterArrivalUpshift = 8;
+    private final static int kInterArrivalShift
+            =  kAbsSendTimeFraction + kAbsSendTimeInterArrivalUpshift;
+    private final static int kInitialProbingIntervalMs = 2000;
+    private final static int kMinClusterSize = 4;
+    private final static int kMaxProbePackets = 15;
+    private final static int kExpectedNumberOfProbes = 3;
+
+    private static final double kTimestampToMs = 1000.0 /
+            (1 << kInterArrivalShift) ;
+
+    private <K,V> List<K> Keys(TreeMap<K,V> _map)
+    {
+        ArrayList<K> keys = new ArrayList<K>();
+        for(Map.Entry<K,V> entry : _map.entrySet())
+        {
+            keys.add(entry.getKey());
+        }
+        return keys;
+    }
+
+    private long ConvertMsTo24Bits(long timeMs)
+    {
+        long time24Bits = (long)(((timeMs << kAbsSendTimeFraction) + 500) /
+                1000) & 0x00FFFFFF;
+        return time24Bits;
+    }
```

Comment:
Method definitions should go below the class fields and constructors (see above)